content/en/blog/_posts/2022-03-31-service-ip-dynamic-and-static-allocation.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: blog
+title: "Avoid collisions assigning IP to Services"
+date: 2022-03-31
+slug: service-ip-dynamic-and-static-allocation
+---
+
+**Author:** Antonio Ojea
+
+
+In Kubernetes, [Services](/docs/concepts/services-networking/service/) are an abstract way to expose
+an application running on a set of Pods. Services
+can have a `ClusterIP` that is virtual and allows to load-balance traffic across the different Pods.
+
+## How Service ClusterIPs are allocated?
+
+A Service `ClusterIP` can be assigned:
+
+_dynamically_
+: the cluster's control plane automatically picks a free IP address from within the configured IP range for `type: ClusterIP` Services.
+
+_statically_
+: you specify an IP address of your choice, from within the configured IP range for Services.
+
+Across your whole cluster, every Service `ClusterIP` must be unique.
+Trying to create a Service with a `ClusterIP` that has already
+been allocated will return an error.
+
+## Why do you need to reserve Service Cluster IPs?
+
+Because you may want to have Services running in well-known IP addresses, so other components and
+users in the cluster can use them.
+
+The best example is the DNS Service for the cluster. Usually, people assign the 10th IP address in
```

Comment:
```suggestion
The best example is the DNS Service for the cluster. Often, cluster operators assign the 10th address from
```
Do we have evidence that this practice happens more than 50% of the time?

Reply:
kubeadm and openshift does it, also the custom scripts to create clusters in kubernetes/kubernetes, but you are right that I should not make that assumption lightly